js/keybindings.js

```diff
@@ -374,15 +369,13 @@ settings.get('keyMap', function (keyMapSettings) {
   defineShortcut('switchToPreviousTask', function (d) {
     taskOverlay.show()
 
-    var currentTaskIdx = tasks.get().map(function (task) {
-      return task.id
-    }).indexOf(currentTask.id)
+    const currentTaskIdx = tasks.indexOf(currentTask),
+          taskCount = tasks.getLength()
 
-    if (tasks.get()[currentTaskIdx - 1]) {
-      browserUI.switchToTask(tasks.get()[currentTaskIdx - 1].id)
-    } else {
-      browserUI.switchToTask(tasks.get()[tasks.get().length - 1].id)
-    }
+    // The addition of taskCount fixes 0 --> -1
+    // E.g. with 5 elements: (0-1+5)%5 = 4, i.e. the last index
+    const previousTaskIndex = (currentTaskIdx - 1 + taskCount) % taskCount
```

Comment:
Can you do something similar here as above, like this?
```
    const previousTask = tasks.byIndex(currentTaskIdx - 1) || tasks.byIndex(tasks.getLength() - 1)
```

Reply:
Oh sorry, didn't see it. Will fix it ASAP.